Requiring `trailing_metadata` is a breaking change

PR #286 added support for showing error status details -- unfortunately, it also [breaks Bigtable's unit tests](https://source.cloud.google.com/results/invocations/539e41cd-9cfd-4db7-9f53-f5da66016909/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fpython-bigtable%2Fpresubmit%2Fpresubmit/log):

```python
___ Test_retry_read_rows_exception.test_w_deadline_exceeded_wrapped_in_grpc ____

self = <tests.unit.test_row_data.Test_retry_read_rows_exception testMethod=test_w_deadline_exceeded_wrapped_in_grpc>

    def test_w_deadline_exceeded_wrapped_in_grpc(self):
        from google.api_core.exceptions import DeadlineExceeded

        wrapped = DeadlineExceeded("testing")
        exception = self._make_grpc_call_error(wrapped)
>       self.assertTrue(self._call_fut(exception))

tests/unit/test_row_data.py:324:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/unit/test_row_data.py:268: in _call_fut
    return _retry_read_rows_exception(exc)
google/cloud/bigtable/row_data.py:333: in _retry_read_rows_exception
    exc = exceptions.from_grpc_error(exc)
.nox/unit-3-9/lib/python3.9/site-packages/google/api_core/exceptions.py:537: in from_grpc_error
    details=_parse_grpc_error_details(rpc_exc),
.nox/unit-3-9/lib/python3.9/site-packages/google/api_core/exceptions.py:490: in _parse_grpc_error_details
    status = rpc_status.from_call(rpc_exc)
.nox/unit-3-9/lib/python3.9/site-packages/grpc_status/rpc_status.py:48: in from_call
    if call.trailing_metadata() is None:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = TestingException(DeadlineExceeded('testing'))

    @abc.abstractmethod
    def trailing_metadata(self):
        """Accesses the trailing metadata sent by the server.

        This method blocks until the value is available.

        Returns:
          The trailing :term:`metadata`.
        """
>       raise NotImplementedError()
E       NotImplementedError

.nox/unit-3-9/lib/python3.9/site-packages/grpc/__init__.py:382: NotImplementedError
__________ Test_retry_read_rows_exception.test_w_miss_wrapped_in_grpc __________

self = <tests.unit.test_row_data.Test_retry_read_rows_exception testMethod=test_w_miss_wrapped_in_grpc>

    def test_w_miss_wrapped_in_grpc(self):
        from google.api_core.exceptions import Conflict

        wrapped = Conflict("testing")
        exception = self._make_grpc_call_error(wrapped)
>       self.assertFalse(self._call_fut(exception))

tests/unit/test_row_data.py:310:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/unit/test_row_data.py:268: in _call_fut
    return _retry_read_rows_exception(exc)
google/cloud/bigtable/row_data.py:333: in _retry_read_rows_exception
    exc = exceptions.from_grpc_error(exc)
.nox/unit-3-9/lib/python3.9/site-packages/google/api_core/exceptions.py:537: in from_grpc_error
    details=_parse_grpc_error_details(rpc_exc),
.nox/unit-3-9/lib/python3.9/site-packages/google/api_core/exceptions.py:490: in _parse_grpc_error_details
    status = rpc_status.from_call(rpc_exc)
.nox/unit-3-9/lib/python3.9/site-packages/grpc_status/rpc_status.py:48: in from_call
    if call.trailing_metadata() is None:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = TestingException(Conflict('testing'))

    @abc.abstractmethod
    def trailing_metadata(self):
        """Accesses the trailing metadata sent by the server.

        This method blocks until the value is available.

        Returns:
          The trailing :term:`metadata`.
        """
>       raise NotImplementedError()
E       NotImplementedError

.nox/unit-3-9/lib/python3.9/site-packages/grpc/__init__.py:382: NotImplementedError
__ Test_retry_read_rows_exception.test_w_service_unavailable_wrapped_in_grpc ___

self = <tests.unit.test_row_data.Test_retry_read_rows_exception testMethod=test_w_service_unavailable_wrapped_in_grpc>

    def test_w_service_unavailable_wrapped_in_grpc(self):
        from google.api_core.exceptions import ServiceUnavailable

        wrapped = ServiceUnavailable("testing")
        exception = self._make_grpc_call_error(wrapped)
>       self.assertTrue(self._call_fut(exception))

tests/unit/test_row_data.py:317:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/unit/test_row_data.py:268: in _call_fut
    return _retry_read_rows_exception(exc)
google/cloud/bigtable/row_data.py:333: in _retry_read_rows_exception
    exc = exceptions.from_grpc_error(exc)
.nox/unit-3-9/lib/python3.9/site-packages/google/api_core/exceptions.py:537: in from_grpc_error
    details=_parse_grpc_error_details(rpc_exc),
.nox/unit-3-9/lib/python3.9/site-packages/google/api_core/exceptions.py:490: in _parse_grpc_error_details
    status = rpc_status.from_call(rpc_exc)
.nox/unit-3-9/lib/python3.9/site-packages/grpc_status/rpc_status.py:48: in from_call
    if call.trailing_metadata() is None:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = TestingException(ServiceUnavailable('testing'))

    @abc.abstractmethod
    def trailing_metadata(self):
        """Accesses the trailing metadata sent by the server.

        This method blocks until the value is available.

        Returns:
          The trailing :term:`metadata`.
        """
>       raise NotImplementedError()
E       NotImplementedError

.nox/unit-3-9/lib/python3.9/site-packages/grpc/__init__.py:382: NotImplementedError
```

--- a/google/api_core/exceptions.py
+++ b/google/api_core/exceptions.py
@@ -487,9 +487,14 @@ def _is_informative_grpc_error(rpc_exc):
 
 
 def _parse_grpc_error_details(rpc_exc):
-    status = rpc_status.from_call(rpc_exc)
+    try:
+        status = rpc_status.from_call(rpc_exc)
+    except NotImplementedError:  # workaround
+        return []
+
     if not status:
         return []
+
     possible_errors = [
         error_details_pb2.BadRequest,
         error_details_pb2.PreconditionFailure,

--- a/tests/unit/test_exceptions.py
+++ b/tests/unit/test_exceptions.py
@@ -237,6 +237,33 @@ def test_from_grpc_error_non_call():
     assert exception.response == error
 
 
+@pytest.mark.skipif(grpc is None, reason="No grpc")
+def test_from_grpc_error_bare_call():
+    message = "Testing"
+
+    class TestingError(grpc.Call, grpc.RpcError):
+        def __init__(self, exception):
+            self.exception = exception
+
+        def code(self):
+            return self.exception.grpc_status_code
+
+        def details(self):
+            return message
+
+    nested_message = "message"
+    error = TestingError(exceptions.GoogleAPICallError(nested_message))
+
+    exception = exceptions.from_grpc_error(error)
+
+    assert isinstance(exception, exceptions.GoogleAPICallError)
+    assert exception.code is None
+    assert exception.grpc_status_code is None
+    assert exception.message == message
+    assert exception.errors == [error]
+    assert exception.response == error
+
+
 def create_bad_request_details():
     bad_request_details = error_details_pb2.BadRequest()
     field_violation = bad_request_details.field_violations.add()

--- a/tests/unit/test_exceptions.py
+++ b/tests/unit/test_exceptions.py
@@ -262,6 +262,7 @@ def test_from_grpc_error_bare_call():
     assert exception.message == message
     assert exception.errors == [error]
     assert exception.response == error
+    assert exception.details == []
 
 
 def create_bad_request_details():

